carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java

```diff
@@ -80,11 +81,17 @@
     private ConfigurationStore store;
     private volatile int period = 0; // in seconds
     private volatile int keyPairsSize = DEFAULT_KEYPAIRS_SIZE;
+    private GroupMembershipHandler groupMembershipHandler;
 
     public void setConfigurationStore(ConfigurationStore configStore) {
         this.store = configStore;
     }
 
+    public void setGroupMembershipHandler(GroupMembershipHandler groupMembershipHandler) {
```

Comment:
this is not a simply "setter"
it is better to rename it to something like 'attachGroupMembershipHandler'